aggie-coding-club/Rev-Registration
# Google social auth now works

## Description

This pull request allows for google authentication when the google login button is pressed in the top right corner of the nav bar. In order for this to work without leaking our client and secret key, y'all are going to have to create a .env file. The instructions are as follows.
1) Navigate to the settings directory inside autoscheduler
2) Create a new file called ".env"
3) Within the .env file, type/copy paste the following.
`GOOGLE_OAUTH2_CLIENT_ID =`
`GOOGLE_OAUTH2_SECRET = `
This picture shows the contents of the .env file.
![tempsnip](https://user-images.githubusercontent.com/16074716/86629418-04358180-bf91-11ea-8ecf-7b058542a774.png)
4) Set the client_id and secret values equal to what is in the discord using single quotes.
ex) `GOOGLE_OAUTH2_CLIENT_ID = 'MAGICVALUE'`
5) Once both values are set, save the changes to your .env file. It should now be properly set up.
## How to test

To test, first run `python manage.py migrate` because the social auth app needs to create some tables. After this is done, click the login button and see if the google login page pops up. Login in with a valid google account. If the authentication is successful, you should be redirected back to the homepage. Any other behavior likely means this isn't working right.

## Screenshots

This is what should appear after pressing the google login button.

![googleauth page](https://user-images.githubusercontent.com/16074716/86050463-84039f00-ba19-11ea-86c4-37c029974fb8.JPG)

## Related tasks

Closes #156
